### PR TITLE
Enforce consistent logic for CreateMetricDescriptor for stats and metrics

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -198,25 +203,26 @@ func (se *statsExporter) createMetricDescriptorFromMetric(ctx context.Context, m
...
+
 	// Otherwise, we encountered a cache-miss and
 	// should create the metric descriptor remotely.
 	inMD, err := se.metricToMpbMetricDescriptor(metric)
 	if err != nil {
 		return err
 	}
 
-	if builtinMetric(inMD.Type) {
+	cmrdesc := &monitoringpb.CreateMetricDescriptorRequest{
```
Comment (songy23) [inline metrics.go]: Consider making a helper function for this few lines so that `metrics`, `metrics_proto` and `stats` can reuse.

Comment (bogdandrutu) [reply metrics.go]: Done.

File: stats.go [hunk 2]
```diff
@@ -64,14 +64,11 @@ type statsExporter struct {
...
-	createdViewsMu sync.Mutex
-	createdViews   map[string]*metricpb.MetricDescriptor // Views already created remotely
-
 	protoMu                sync.Mutex
-	protoMetricDescriptors map[string]bool // Saves the metric descriptors that were already created remotely
+	protoMetricDescriptors map[string]bool // Metric descriptors that were already created remotely
 
 	metricMu          sync.Mutex
-	metricDescriptors map[string]bool // Saves the metric descriptors that were already created remotely
+	metricDescriptors map[string]bool // <etric descriptors that were already created remotely
```
Comment (rghetia) [inline stats.go]: nit: s/<etric/Metric/

File: stats.go [hunk 3]
```diff
@@ -335,44 +331,37 @@ func (e *statsExporter) viewToMetricDescriptor(ctx context.Context, v *view.View
 // An error will be returned if there is already a metric descriptor created with the same name
 // but it has a different aggregation or keys.
 func (e *statsExporter) createMeasure(ctx context.Context, v *view.View) error {
```
Comment (rghetia) [inline stats.go]: nit: May be createMeasure should be renamed to createMetricDescriptorFromView like others

File: stats.go [hunk 4]
```diff
@@ -335,44 +331,37 @@ func (e *statsExporter) viewToMetricDescriptor(ctx context.Context, v *view.View
...
+	defer e.metricMu.Unlock()
 
 	viewName := v.Name
 
-	if md, ok := e.createdViews[viewName]; ok {
-		// [TODO:rghetia] Temporary fix for https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/issues/76#issuecomment-459459091
-		if builtinMetric(md.Type) {
-			return nil
-		}
-		return e.equalMeasureAggTagKeys(md, v.Measure, v.Aggregation, v.TagKeys)
```
Comment (rghetia) [inline stats.go]: is this check not required anymore?

Comment (bogdandrutu) [reply stats.go]: This is not used in metrics or metrics_proto. I think we tried to do too much in the client. If the metric changed the definition user should deal with this by changing the name. We cannot correct this error.

File: stats.go (hunk 2 above)
Comment (bogdandrutu) [reply stats.go]: done.

File: stats.go (hunk 3 above)
Comment (bogdandrutu) [reply stats.go]: Done.

File: stats.go (hunk 4 above)
Comment (rghetia) [reply stats.go]: SG

**Pull request merged**